src/components/AudioPlayer/PlayPauseButton.tsx

```diff
@@ -1,24 +1,47 @@
-import { shallowEqual, useSelector } from 'react-redux';
+import { useState } from 'react';
+import { shallowEqual, useDispatch, useSelector } from 'react-redux';
+import useChaptersIdsByUrlPath from 'src/hooks/useChapterId';
 import {
   AudioFileStatus,
+  loadAndPlayAudioFile,
+  selectAudioFile,
   selectAudioFileStatus,
   selectAudioPlayerState,
 } from 'src/redux/slices/AudioPlayer/state';
+import { getChapterData } from 'src/utils/chapter';
 import { withStopPropagation } from 'src/utils/event';
 
 import PauseIcon from '../../../public/icons/pause.svg';
 import PlayIcon from '../../../public/icons/play-arrow.svg';
 import Button, { ButtonShape, ButtonSize, ButtonVariant } from '../dls/Button/Button';
 import Spinner, { SpinnerSize } from '../dls/Spinner/Spinner';
 import { triggerPauseAudio, triggerPlayAudio } from './EventTriggers';
+import SurahAudioMismatchModal from './SurahAudioMismatchModal';
 
 const PlayPauseButton = () => {
+  const dispatch = useDispatch();
+
   const { isPlaying } = useSelector(selectAudioPlayerState, shallowEqual);
-  const audioFileStatus = useSelector(selectAudioFileStatus);
-  const isLoading = audioFileStatus === AudioFileStatus.Loading;
+  const isLoading = useSelector(selectAudioFileStatus) === AudioFileStatus.Loading;
```

Comment:
should we add `shallowEqual` here? if this PR doesn't go before the `RootState` one.

Reply:
yep I think this PR will go before RootState. 

For audioFileStatus I think this is okay, it will transpiled into primitive (tested with typescript repl.it) CMIIW